ui-admin/src/study/kits/KitList.tsx

```diff
@@ -0,0 +1,122 @@
+import { instantToDateString, isoToInstant } from 'util/timeUtils'
+
+const pepperStatusToHumanStatus = (dsmStatus?: string): string => {
+  const statusMap: Record<string, string> = {
+    'CREATED': 'Created',
+    'LABELED': 'Prepared',
+    'SCANNED': 'Sent',
+    'RECEIVED': 'Returned'
+  }
+  return dsmStatus
+    ? (statusMap[dsmStatus] || (`(${  dsmStatus  })`))
+    : '(unknown)'
+}
+
+/** Loads sample kits for a study and shows them as a list. */
+export default function KitList({ studyEnvContext }: { studyEnvContext: StudyEnvContextT }) {
+  const { portal, study, currentEnv } = studyEnvContext
+  const [isLoading, setIsLoading] = useState(true)
+  const [kits, setKits] = useState<KitRequest[]>([])
+
+  const loadKits = async () => {
+    setIsLoading(true)
+    const kits = await Api.fetchKitsByStudyEnvironment(
+      portal.shortcode, study.shortcode, currentEnv.environmentName)
+    setKits(kits)
+    setIsLoading(false)
+  }
+
+  useEffect(() => {
+    loadKits()
+  }, [])
+
+  return <LoadingSpinner isLoading={isLoading}>
+    <KitListView kits={kits} studyEnvContext={studyEnvContext}/>
+  </LoadingSpinner>
+}
+
+/** Renders a table with a list of kits. */
+export function KitListView({ studyEnvContext, kits }: { studyEnvContext: StudyEnvContextT, kits: KitRequest[] }) {
+  const { currentEnvPath } = studyEnvContext
+
+  const countsByStatus = _countBy(kits, kit => kit.pepperStatus?.currentStatus || '')
+
+  const columns: ColumnDef<KitRequest, string>[] = [{
+    header: 'Enrollee shortcode',
+    accessorKey: 'enrollee.shortcode',
+    meta: {
+      columnType: 'string'
+    },
+    cell: data => <Link to={`${currentEnvPath}/participants/${data.getValue()}`}>{data.getValue()}</Link>,
+    enableColumnFilter: false
+  }, {
+    header: 'Kit type',
+    accessorKey: 'kitType.displayName',
+    enableColumnFilter: false
+  }, {
+    header: 'Created',
+    accessorKey: 'createdAt',
+    cell: data => instantToDateString(Number(data.getValue())),
+    enableColumnFilter: false
+  }, {
+    header: 'Prepared',
+    accessorKey: 'pepperStatus.labelDate',
+    cell: data => instantToDateString(isoToInstant(data.getValue())),
+    enableColumnFilter: false
+  }, {
+    header: 'Sent',
+    accessorKey: 'pepperStatus.scanDate',
+    cell: data => instantToDateString(isoToInstant(data.getValue())),
+    enableColumnFilter: false
+  }, {
+    header: 'Returned',
+    accessorKey: 'pepperStatus.receiveDate',
+    cell: data => instantToDateString(isoToInstant(data.getValue())),
+    enableColumnFilter: false
+  }, {
+    header: 'Status',
+    accessorKey: 'pepperStatus.currentStatus',
+    accessorFn: data => pepperStatusToHumanStatus(data?.pepperStatus?.currentStatus),
+    meta: {
+      columnType: 'string',
+      filterOptions: [
+        { value: true, label: 'Created' },
+        { value: true, label: 'Prepared' },
+        { value: true, label: 'Sent' },
+        { value: true, label: 'Received' }
+      ]
```

Comment:
I don't think this column filter is working as you're intending. Currently it shows up as a search bar where you can type in the status, but given the use of `filterOptions` here I am assuming you want a dropdown. When I wrote the code for the custom dropdown filter, I only implemented it for `boolean`. 

I just pushed this commit to another branch that extends it to handle `string` columns as well, if you'd like to pull it into this PR: https://github.com/broadinstitute/juniper/commit/49a607428d4d7e0462f7753e000cbec5f2869d7c

Reply:
Oh, nice! Yes, this is a place where I had to make a decision to let it be imperfect rather than spend more time on it. This _might_ be less relevant depending on Erin's direction with designs, but I'll definitely update it now if it's easy to do.